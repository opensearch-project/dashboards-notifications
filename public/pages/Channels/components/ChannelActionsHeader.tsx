/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiButton, EuiContextMenuItem, EuiPopover } from '@elastic/eui';
import React, { useState } from 'react';
import { ChannelItemType } from '../../../../models/interfaces';
import { ModalConsumer } from '../../../components/Modal';
import { ROUTES } from '../../../utils/constants';
import { DeleteChannelModal } from './modals/DeleteChannelModal';

interface ChannelActionsParams {
  label: string;
  disabled: boolean;
  modal?: React.ReactNode;
  modalParams?: object;
  href?: string;
  action?: () => void;
}

interface ChannelActionsProps {
  selected: ChannelItemType[];
  setSelected: (items: ChannelItemType[]) => void;
  items: ChannelItemType[];
  setItems: (items: ChannelItemType[]) => void;
  refresh: () => void;
  href?: string;
}

export function ChannelActionsHeader(props: ChannelActionsProps) {
  const [isPopoverOpen, setIsPopoverOpen] = useState(false);

  const actions: ChannelActionsParams[] = [
    {
      label: 'Edit',
      disabled: props.selected.length !== 1,
      href: props.href,
    },
    {
      label: 'Delete',
      disabled: props.selected.length === 0,
      modal: DeleteChannelModal,
      modalParams: { refresh: props.refresh },
    },
  ];

  return (
    <ModalConsumer>
      {({ onShow }) => (
        <EuiPopover
          panelPaddingSize="none"
          button={
            <EuiButton
              iconType="arrowDown"
              iconSide="right"
              disabled={props.selected.length === 0}
              onClick={() => setIsPopoverOpen(!isPopoverOpen)}
            >
              Actions
            </EuiButton>
          }
          isOpen={isPopoverOpen}
          closePopover={() => setIsPopoverOpen(false)}
        >
          {actions.map((params) => (
            <EuiContextMenuItem
              key={params.label}
              disabled={params.disabled}
              onClick={() => {
                setIsPopoverOpen(false);
                if (params.modal) {
                  onShow(params.modal, {
                    selected: props.selected,
                    ...(params.modalParams || {}),
                  });
                }
                if (params.href) location.assign(params.href);
                if (params.action) params.action();
              }}
            >
              {params.label}
            </EuiContextMenuItem>
          ))}
        </EuiPopover>
      )}
    </ModalConsumer>
  );
}
