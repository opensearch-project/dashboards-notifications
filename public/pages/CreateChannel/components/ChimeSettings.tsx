/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiCompressedFieldText, EuiCompressedFormRow } from '@elastic/eui';
import React, { useContext } from 'react';
import { CreateChannelContext } from '../CreateChannel';
import { validateWebhookURL } from '../utils/validationHelper';

interface ChimeSettingsProps {
  chimeWebhook: string;
  setChimeWebhook: (url: string) => void;
}

export function ChimeSettings(props: ChimeSettingsProps) {
  const context = useContext(CreateChannelContext)!;

  return (
    <EuiCompressedFormRow
      label="Webhook URL"
      style={{ maxWidth: '700px' }}
      error={context.inputErrors.chimeWebhook.join(' ')}
      isInvalid={context.inputErrors.chimeWebhook.length > 0}
    >
      <EuiCompressedFieldText
        fullWidth
        data-test-subj="create-channel-chime-webhook-input"
        placeholder="https://hooks.chime.aws/incomingwebhooks/XXXXX..."
        value={props.chimeWebhook}
        onChange={(e) => props.setChimeWebhook(e.target.value)}
        isInvalid={context.inputErrors.chimeWebhook.length > 0}
        onBlur={() => {
          context.setInputErrors({
            ...context.inputErrors,
            chimeWebhook: validateWebhookURL(props.chimeWebhook),
          });
        }}
      />
    </EuiCompressedFormRow>
  );
}
