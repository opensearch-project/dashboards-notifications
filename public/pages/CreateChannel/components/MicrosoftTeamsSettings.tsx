/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiFieldText, EuiFormRow } from '@elastic/eui';
import React, { useContext } from 'react';
import { CreateChannelContext } from '../CreateChannel';
import { validateWebhookURL } from '../utils/validationHelper';

interface MicrosoftTeamsSettingsProps {
  microsoftTeamsWebhook: string;
  setMicrosoftTeamsWebhook: (url: string) => void;
}

export function MicrosoftTeamsSettings(props: MicrosoftTeamsSettingsProps) {
  const context = useContext(CreateChannelContext)!;

  return (
    <EuiFormRow
      label="Webhook URL"
      style={{ maxWidth: '700px' }}
      error={context.inputErrors.microsoftTeamsWebhook.join(' ')}
      isInvalid={context.inputErrors.microsoftTeamsWebhook.length > 0}
    >
      <EuiFieldText
        fullWidth
        data-test-subj="create-channel-microsoftTeams-webhook-input"
        placeholder="https://xxxxx.webhook.office.com/webhookb2/xxxxx/IncomingWebhook/xxxxx..."
        value={props.microsoftTeamsWebhook}
        onChange={(e) => props.setMicrosoftTeamsWebhook(e.target.value)}
        isInvalid={context.inputErrors.microsoftTeamsWebhook.length > 0}
        onBlur={() => {
          context.setInputErrors({
            ...context.inputErrors,
            microsoftTeamsWebhook: validateWebhookURL(props.microsoftTeamsWebhook),
          });
        }}
      />
    </EuiFormRow>
  );
}
