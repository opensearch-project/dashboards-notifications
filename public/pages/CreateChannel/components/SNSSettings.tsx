/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiCallOut,
  EuiFieldText,
  EuiFormRow,
  EuiLink,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';
import React, { useContext } from 'react';
import { DOCUMENTATION_LINK } from '../../../utils/constants';
import { MainContext } from '../../Main/Main';
import { CreateChannelContext } from '../CreateChannel';
import { validateArn } from '../utils/validationHelper';
import { i18n } from '@osd/i18n';

interface SNSSettingsProps {
  topicArn: string;
  setTopicArn: (topicArn: string) => void;
  roleArn: string;
  setRoleArn: (roleArn: string) => void;
}

export function SNSSettings(props: SNSSettingsProps) {
  const context = useContext(CreateChannelContext)!;
  const mainStateContext = useContext(MainContext)!;
  return (
    <>
      <EuiSpacer />
      <EuiFormRow
        label={i18n.translate('notification.notificationChannels.snsTopicArn', {
          defaultMessage:
          "SNS topic ARN",
          })}
        error={context.inputErrors.topicArn.join(' ')}
        isInvalid={context.inputErrors.topicArn.length > 0}
      >
        <EuiFieldText
          placeholder={i18n.translate('notification.notificationChannels.arnKey', {
            defaultMessage:
            "ARN key",
            })}
          data-test-subj="sns-settings-topic-arn-input"
          value={props.topicArn}
          onChange={(e) => props.setTopicArn(e.target.value)}
          isInvalid={context.inputErrors.topicArn.length > 0}
          onBlur={() => {
            context.setInputErrors({
              ...context.inputErrors,
              topicArn: validateArn(props.topicArn),
            });
          }}
        />
      </EuiFormRow>

      {mainStateContext.tooltipSupport ? (
        <>
          <EuiFormRow
            label={
              <span>
                {i18n.translate('notification.notificationChannels.iamRoleArn', {
                  defaultMessage:
                    'IAM role ARN - ',
                  })}
                <i style={{ fontWeight: 'normal' }}>
                {i18n.translate('notification.notificationChannels.newChannelDescriptionOptional', {
                defaultMessage:
                  'optional',
                })}
                </i>
              </span>
            }
          >
            <>
              <EuiText size="xs" color="subdued">
              {i18n.translate('notification.notificationChannels.iamRoleDesc', {
                defaultMessage:
                  'IAM role ARN can only be used for clusters running on AWS network.',
                })}
                
              </EuiText>
              <EuiSpacer size="s" />
              <EuiFieldText
                data-test-subj="sns-settings-role-arn-input"
                placeholder={i18n.translate('notification.notificationChannels.arnKey', {
                  defaultMessage:
                  "ARN key",
                  })}
                value={props.roleArn}
                onChange={(e) => props.setRoleArn(e.target.value)}
              />
            </>
          </EuiFormRow>
          <EuiSpacer />
          <EuiCallOut
            title={i18n.translate('notification.notificationChannels.amazonSNS', {
              defaultMessage:
              "Using Amazon SNS outside of AWS",
              })}
            style={{ maxWidth: 720 }}
          >
            <div>
            {i18n.translate('notification.notificationChannels.awsClusterRunningDesc', {
            defaultMessage:
              'If your cluster is not running on AWS, you must configure aws credentials on your OpenSearch cluster.',
            })}

              <EuiLink href={DOCUMENTATION_LINK} target="_blank" external>
              {i18n.translate('notification.notificationChannels.readMore', {
              defaultMessage:
                ' Learn more',
              })}

              </EuiLink>
            </div>
          </EuiCallOut>
        </>
      ) : (
        <EuiFormRow
          label={i18n.translate('notification.notificationChannels.iamRoleArn', {
                defaultMessage:
                "IAM role ARN",
                })}
          error={context.inputErrors.roleArn.join(' ')}
          isInvalid={context.inputErrors.roleArn.length > 0}
        >
          <EuiFieldText
            placeholder={i18n.translate('notification.notificationChannels.arnKey', {
              defaultMessage:
              "ARN key",
              })}
            data-test-subj="sns-settings-role-arn-input"
            value={props.roleArn}
            onChange={(e) => props.setRoleArn(e.target.value)}
            isInvalid={context.inputErrors.roleArn.length > 0}
            onBlur={() => {
              context.setInputErrors({
                ...context.inputErrors,
                roleArn: validateArn(props.roleArn),
              });
            }}
          />
        </EuiFormRow>
      )}
    </>
  );
}
