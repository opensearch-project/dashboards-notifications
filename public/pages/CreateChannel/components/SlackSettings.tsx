/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiCompressedFieldText, EuiCompressedFormRow } from '@elastic/eui';
import React, { useContext } from 'react';
import { CreateChannelContext } from '../CreateChannel';
import { validateWebhookURL } from '../utils/validationHelper';

interface SlackSettingsProps {
  slackWebhook: string;
  setSlackWebhook: (url: string) => void;
}

export function SlackSettings(props: SlackSettingsProps) {
  const context = useContext(CreateChannelContext)!;

  return (
    <EuiCompressedFormRow
      label="Slack webhook URL"
      style={{ maxWidth: '700px' }}
      error={context.inputErrors.slackWebhook.join(' ')}
      isInvalid={context.inputErrors.slackWebhook.length > 0}
    >
      <EuiCompressedFieldText
        fullWidth
        data-test-subj="create-channel-slack-webhook-input"
        placeholder="https://hooks.slack.com/services/XXXXX/XXXXX/XXXXX..."
        value={props.slackWebhook}
        onChange={(e) => props.setSlackWebhook(e.target.value)}
        isInvalid={context.inputErrors.slackWebhook.length > 0}
        onBlur={() => {
          context.setInputErrors({
            ...context.inputErrors,
            slackWebhook: validateWebhookURL(props.slackWebhook),
          });
        }}
      />
    </EuiCompressedFormRow>
  );
}
