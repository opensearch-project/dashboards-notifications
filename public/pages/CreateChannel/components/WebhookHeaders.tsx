/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiSmallButton,
  EuiCompressedFieldText,
  EuiFlexGroup,
  EuiFlexItem,
  EuiCompressedFormRow,
  EuiSpacer,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import React from 'react';
import { HeaderItemType } from '../../Channels/types';

interface WebhookHeadersProps {
  headers: HeaderItemType[];
  setHeaders: (headers: HeaderItemType[]) => void;
  type: 'header' | 'parameter';
}

export function WebhookHeaders(props: WebhookHeadersProps) {
  const setHeader = (
    key: string | null,
    value: string | null,
    index: number
  ) => {
    const header = props.headers[index];
    const newHeaders = [...props.headers];
    if (key !== null) header.key = key;
    else if (value !== null) header.value = value;
    newHeaders.splice(index, 1, header);
    props.setHeaders(newHeaders);
  };

  return (
    <>
      <EuiTitle size="xs">
        <h4>
          {props.type === 'parameter' ? 'Query parameters' : 'Webhook headers'}
        </h4>
      </EuiTitle>

      {props.headers.length === 0 && (
        <>
          <EuiSpacer size="m" />
          <EuiText size="s">{`No ${props.type}s defined.`}</EuiText>
        </>
      )}

      {props.headers.map((header, i) => {
        return (
          <div key={`webhook-${props.type}-${i}`}>
            <EuiSpacer size="s" />
            <EuiFlexGroup style={{ maxWidth: 639 }}>
              <EuiFlexItem>
                <EuiCompressedFormRow label="Key">
                  <EuiCompressedFieldText
                    placeholder=""
                    value={header.key}
                    onChange={(e) => setHeader(e.target.value, null, i)}
                    disabled={
                      props.type === 'header' &&
                      i === 0 &&
                      header.key === 'Content-Type'
                    } // first header needs to be Content-Type
                  />
                </EuiCompressedFormRow>
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCompressedFormRow label="Value">
                  <EuiCompressedFieldText
                    placeholder=""
                    value={header.value}
                    onChange={(e) => setHeader(null, e.target.value, i)}
                  />
                </EuiCompressedFormRow>
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCompressedFormRow hasEmptyLabelSpace>
                  <EuiSmallButton
                    onClick={() => {
                      const newHeaders = [...props.headers];
                      newHeaders.splice(i, 1);
                      props.setHeaders(newHeaders);
                    }}
                    disabled={
                      props.type === 'header' &&
                      i === 0 &&
                      header.key === 'Content-Type'
                    }
                  >
                    {`Remove ${props.type}`}
                  </EuiSmallButton>
                </EuiCompressedFormRow>
              </EuiFlexItem>
            </EuiFlexGroup>
          </div>
        );
      })}

      <EuiSpacer size="m" />
      <EuiSmallButton
        onClick={() => {
          props.setHeaders([...props.headers, { key: '', value: '' }]);
        }}
      >
        {`Add ${props.type}`}
      </EuiSmallButton>
    </>
  );
}
