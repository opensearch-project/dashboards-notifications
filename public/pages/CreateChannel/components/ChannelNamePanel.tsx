/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiFieldText, EuiFormRow, EuiTextArea } from '@elastic/eui';
import React, { useContext } from 'react';
import { ContentPanel } from '../../../components/ContentPanel';
import { CreateChannelContext } from '../CreateChannel';
import { validateChannelName } from '../utils/validationHelper';
import { i18n } from '@osd/i18n';

interface ChannelNamePanelProps {
  name: string;
  setName: (name: string) => void;
  description: string;
  setDescription: (description: string) => void;
}

export function ChannelNamePanel(props: ChannelNamePanelProps) {
  const context = useContext(CreateChannelContext)!;
  return (
    <>
      <ContentPanel
        bodyStyles={{ padding: 'initial' }}
        title={
          i18n.translate('notification.notificationChannels.nameAndDescription', {
          defaultMessage:
          "Name and description",
          })}
        
        titleSize="s"
      >
        <EuiFormRow
          label={
            i18n.translate('notification.notificationChannels.newChannelName', {
            defaultMessage:
              'Name',
            })}
          
          error={context.inputErrors.name.join(' ')}
          isInvalid={context.inputErrors.name.length > 0}
        >
          <EuiFieldText
            data-test-subj="create-channel-name-input"
            placeholder={
              i18n.translate('notification.notificationChannels.newChannelPlaceholder', {
              defaultMessage:
                "Enter channel name",
              })}
            
            value={props.name}
            onChange={(e) => props.setName(e.target.value)}
            isInvalid={context.inputErrors.name.length > 0}
            onBlur={() => {
              context.setInputErrors({
                ...context.inputErrors,
                name: validateChannelName(props.name),
              });
            }}
          />
        </EuiFormRow>
        <EuiFormRow
          label={
            <span>
              {
              i18n.translate('notification.notificationChannels.newChannelDescription', {
              defaultMessage:
                "Description - ",
              })}
              <i style={{ fontWeight: 'normal' }}>{
                i18n.translate('notification.notificationChannels.newChannelDescriptionOptional', {
                defaultMessage:
                  'optional',
                })}
                </i>
            </span>
          }
        >
          <>
            <EuiTextArea
              data-test-subj="create-channel-description-input"
              placeholder={
                i18n.translate('notification.notificationChannels.newChannelPurpose', {
                defaultMessage:
                "What is the purpose of this channel?",
                })}
              
              style={{ height: '4.1rem' }}
              value={props.description}
              onChange={(e) => props.setDescription(e.target.value)}
            />
          </>
        </EuiFormRow>
      </ContentPanel>
    </>
  );
}
