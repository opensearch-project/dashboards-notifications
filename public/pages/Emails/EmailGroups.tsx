/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiSpacer, EuiTitle } from '@elastic/eui';
import React, { useContext, useEffect } from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { CoreServicesContext } from '../../components/coreServices';
import { BREADCRUMBS } from '../../utils/constants';
import { RecipientGroupsTable } from './components/tables/RecipientGroupsTable';
import { MainContext } from '../Main/Main';
import { NotificationService } from '../../services';
import { i18n } from '@osd/i18n';

interface EmailGroupsProps extends RouteComponentProps {
  notificationService: NotificationService;
}

export function EmailGroups(props: EmailGroupsProps) {
  const coreContext = useContext(CoreServicesContext)!;
  useEffect(() => {
    coreContext.chrome.setBreadcrumbs([
      BREADCRUMBS.NOTIFICATIONS,
      BREADCRUMBS.EMAIL_GROUPS,
    ]);
    window.scrollTo(0, 0);
  }, []);

  return (
    <>
      <EuiTitle size="l">
      <h1> 
          {
          i18n.translate('notification.notificationChannels.emailRecipientsGroupsList', {
          defaultMessage:
          'Email recipient groups',
        })}
    </h1>

      </EuiTitle>

      <EuiSpacer />
      <RecipientGroupsTable coreContext={coreContext} notificationService={props.notificationService} />
    </>
  );
}
