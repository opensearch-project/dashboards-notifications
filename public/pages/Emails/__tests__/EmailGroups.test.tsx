/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { render } from '@testing-library/react';
import React from 'react';
import { routerComponentPropsMock } from '../../../../test/mocks/routerPropsMock';
import {
  coreServicesMock,
  notificationServiceMock,
} from '../../../../test/mocks/serviceMock';
import { CoreServicesContext } from '../../../components/coreServices';
import { ServicesContext } from '../../../services';
import { EmailGroups } from '../EmailGroups';

describe('<EmailGroups/> spec', () => {
  it('renders the component', () => {
    const routerComponentPropsMock = {
      // Mock other props as needed
      notificationService: notificationServiceMock,
    };

    const utils = render(
        <CoreServicesContext.Provider value={coreServicesMock}>
          <EmailGroups {...routerComponentPropsMock} />
        </CoreServicesContext.Provider>
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
