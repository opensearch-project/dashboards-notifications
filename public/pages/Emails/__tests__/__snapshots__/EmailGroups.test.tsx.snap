// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailGroups/> spec renders the component 1`] = `
<div
  class="euiText euiText--small"
>
  <h1>
    Email recipient groups
  </h1>
</div>
`;
